File: packages/ts-sdk/src/treedb/adapters/federation-internal.ts
import { jsonRequest, segment, type TreeDbAdapterContext } from './common.js';

export class FederationInternalAdapter {
  constructor(private readonly context: TreeDbAdapterContext) {}
  health(): Promise<unknown> { return jsonRequest(this.context.transport, 'GET', '/api/v1/internal/federation/health'); }
  proxy(input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', '/api/v1/internal/federation/proxy', input); }
  exportMirror(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/internal/federation/repos/${segment(repoId)}/mirror/export`, input); }
  importMirror(repoId: string, input: unknown): Promise<unknown> { return jsonRequest(this.context.transport, 'POST', `/api/v1/internal/federation/repos/${segment(repoId)}/mirror/import`, input); }
}

File: packages/elixir-sdk/lib/tree_db_sdk/adapters/federation_internal.ex
defmodule TreeDbSdk.FederationInternal do
  @moduledoc false
  alias TreeDbSdk.Adapters.Common

  def health(client), do: Common.json_request(client, :get, "/api/v1/internal/federation/health")

  def proxy(client, body),
    do: Common.json_request(client, :post, "/api/v1/internal/federation/proxy", body)

  def export_mirror(client, repo_id, body),
    do:
      Common.json_request(
        client,
        :post,
        "/api/v1/internal/federation/repos/" <> Common.segment(repo_id) <> "/mirror/export",
        body
      )

  def import_mirror(client, repo_id, body),
    do:
      Common.json_request(
        client,
        :post,
        "/api/v1/internal/federation/repos/" <> Common.segment(repo_id) <> "/mirror/import",
        body
      )
end
